src/meta/src/hummock/compaction/mod.rs

```diff
@@ -191,14 +191,26 @@ pub fn create_compaction_task(
     base_level: usize,
     compaction_task_type: compact_task::TaskType,
 ) -> CompactionTask {
-    let target_file_size = if input.target_level == 0 {
-        compaction_config.target_file_size_base
-    } else if input.target_level == base_level {
-        // This is just a temporary optimization measure. We hope to reduce the size of SST as much
-        // as possible to reduce the amount of data blocked by a single task during compaction,
-        // but too many files will increase computing overhead.
-        // TODO: remove it after can reduce configuration `target_file_size_base`.
-        compaction_config.target_file_size_base / 4
+    let target_file_size = if input.target_level == 0 || input.target_level == base_level {
+        let task_size = input.target_input_size + input.select_input_size;
+        if input.target_level > 0 && task_size < compaction_config.max_bytes_for_level_base {
```

Comment:
IMO, It is sporadic that task size is smaller than max_bytes_for_level_base. So the original behavior is changed here.

my suggestion is
1. It is necessary to test daily nexmark and compare the performance.

2. Cover through UT to ensure that the output meets expectations

3. Trigger the branch through the special test mentioned in the PR.